### ISSUE TITLE
Reorganize package structure to support specific driver versions

### ISSUE THREAD

**Issue by thobbs** [riptano/cassandra-dtest#188]
We've been working on some protocol changes for the v4 native protocol.  Tests for the native protocol generally require a driver, but the python driver doesn't have support for the new features yet.  So, to properly write tests, we need to run with a forked version of the python driver.

I attempted to make this possible with a test-level decorator, but it's too tricky to replace/reload a package completely in python.

I think the next best option might be a combination of two things:
* An environment variable to specify a repo and branch of the python driver to use for the tests. I already have code to handle a git clone and checkout of the driver, given a repository and branch name.
* A test decorator to skip the test unless that version of the python driver is in use

The main change that would be required is to create a proper package structure for the dtests.  That probably looks like a ``dtests`` directory with the test modules in it, along with an ``__init__.py`` file.  The ``__init__.py`` file is key, because it's loaded before any of the test modules.  That's where the environment variable would be checked, allowing us to import the python driver from a specific location.  If we do this anywhere else, there's no way to guarantee that the driver package hasn't already been imported.

In the short-term, this would cause some pain around updating test runners and imports.  In the long-run, though, a proper package structure should make it easier to do things like this (and hopefully make the code a little cleaner).

What do you guys think?

**Comment by knifewine** [comment]
+1 as long as we can keep the functionality unsurprising for everyday uses of dtest (such as using nose locally to run a handful of tests).

on the topic of reorg I think we should consider moving test modules into separate functional area subpackages (subjective, yes) so there's room for multiple modules within a given functional area, and helper code can have its own module if desired.

**Comment by thobbs** [comment]
bq. +1 as long as we can keep the functionality unsurprising for everyday uses of dtest (such as using nose locally to run a handful of tests).

Yup, it would remain easy to use nose.  The main difference would be pointing nose at a directory (although I think it might work without a directory, too).

bq. I think we should consider moving test modules into separate functional area subpackages

I'm fine with doing this in my changes, but I don't have any strong opinions on what the layout should be.  Feel free to suggest something if you'd like me to make those changes in my PR.

**Comment by ptnapoleon** [comment]
I don't think we need to make you do additional organization of dtest. As long as your changes are future-compatible, i.e., we don't need to rework your changes when we do org dtest, then that should be sufficient.

**Comment by ptnapoleon** [comment]
@thobbs Seems like we are +1 on this.
